core/src/main/java/fr/sncf/osrd/new_infra/api/reservation/ReservationRoute.java

```diff
@@ -0,0 +1,14 @@
+package fr.sncf.osrd.new_infra.api.reservation;
+
+import com.google.common.collect.ImmutableList;
+
+public interface ReservationRoute {
+    /** Returns the path the route takes in the detector graph */
+    ImmutableList<DiDetector> getDetectorPath();
```

Comment:
It's an edge between `DiDetectors`, does it make sense to have a list of detectors? This could mess up the graph and pathfinding if an edge includes a detector without it being registered as so in the graph